### PR TITLE
Admin page

### PR DESCRIPTION
**Pull request by Rafer45**
Completed the admin control panel page!

Pagination is kind of urgent for this and people cannot be demoted (or deadminified), but this is a good stage -- admins can now be added!

### CODE REVIEW

File: src/test/java/codeu/controller/ControlPanelServletTest.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("nobody");
+
+    controlPanelServlet.doPost(mockRequest, mockResponse);
+
+
+  }
+
+  @Test
+  public void testDoPost_NotAdminUsername() throws IOException, ServletException {
+    ;
```
Comment (eselenic) [inline src/test/java/codeu/controller/ControlPanelServletTest.java]: Are these suppose to be empty??

Comment (eselenic) [reply src/test/java/codeu/controller/ControlPanelServletTest.java]: If so, I would add a comment TODO just so it's more understandable.

File: src/test/java/codeu/controller/ControlPanelServletTest.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
```
Comment (jcolladokuri) [inline src/test/java/codeu/controller/ControlPanelServletTest.java]: Add TODO or code to these two test cases

**Review by jcolladokuri — CHANGES_REQUESTED** [review]

File: src/main/java/codeu/model/data/User.java [hunk 3]
```diff
@@ -25,7 +25,7 @@
   private final String passwordHash;
   private final Instant creation;
   private boolean isAdmin;
-  private Set<UUID> conversationSet;
+  private HashSet<UUID> conversationSet;
```
Comment (jcolladokuri) [inline src/main/java/codeu/model/data/User.java]: why did you change from Set to HashSet? Just trying to understand the logic behind it

File: src/test/java/codeu/controller/ControlPanelServletTest.java (hunk 1 above)
Comment (Rafer45) [reply src/test/java/codeu/controller/ControlPanelServletTest.java]: Done!

File: src/test/java/codeu/controller/ControlPanelServletTest.java (hunk 2 above)
Comment (Rafer45) [reply src/test/java/codeu/controller/ControlPanelServletTest.java]: Done!

File: src/main/java/codeu/model/data/User.java (hunk 3 above)
Comment (Rafer45) [reply src/main/java/codeu/model/data/User.java]: I was going to use `Set` to be more generic, but it turns out that the `Set` interface isn't serializable,. Since it's not guaranteed to be serializable, it's not guaranteed to have the ability to turn into raw binary for persistent storage. So, I had to use a specific implementation of `Set` that is serializable.

Comment (jcolladokuri) [reply src/main/java/codeu/model/data/User.java]: got it, thanks!

**Review by jcolladokuri — APPROVED** [review]
